cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -109,6 +109,12 @@ if("${jemalloc_SOURCE}" STREQUAL "")
   set(jemalloc_SOURCE "BUNDLED")
 endif()
 
+# GH-36013 bundled opentelemetry-cpp build is broken, change the default
+# to CONDA until a patched bundled version is available.
+if("${opentelemetry-cpp_SOURCE}" STREQUAL "")
+  set(opentelemetry-cpp_SOURCE "CONDA")
+endif()
+
```

Comment:
Do we need this?
I think that this is needless. I think that we just need to disable OpenTelemetry without conda.